chore: OPTIC-1935: Generate all required css variables from design tokens

bmartel opened the pull request:
<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->
This pull request introduces a new theme toggle feature for Storybook and updates the color tokens to include raw values. The most important changes include adding the theme toggle addon, updating Storybook configuration files, and modifying the color tokens.

### Theme Toggle Feature:

* [`web/libs/storybook/.storybook/main.ts`](diffhunk://#diff-6f63f4116d05a8eec64d5a254f51c43d696525dcbe8dc32f58065b91656d54f8R13): Added the theme toggle addon to the list of Storybook addons.
* [`web/libs/storybook/addons/theme-toggle/`](diffhunk://#diff-abc95cac21ae24a55bb9a66a71a34920faaaa774505190cf9be04edeb3e5686bR1-R33): Implemented the theme toggle feature with new files for atoms, constants, preview, register, and theme-toggle components. [[1]](diffhunk://#diff-abc95cac21ae24a55bb9a66a71a34920faaaa774505190cf9be04edeb3e5686bR1-R33) [[2]](diffhunk://#diff-21235ae48fd329994a85c7fd4d583e77ff82236cfd139aeef39b7c0812d1f6afR1-R5) [[3]](diffhunk://#diff-c23737a72d49ae1507df85b06d8aad342beae8ed54e082a4dd5406f11c2033e8R1-R13) [[4]](diffhunk://#diff-3ee7b6e50b4c0e242d5879e73c1c21aedda208375e72486a59e3ef57702f03b7R1-R12) [[5]](diffhunk://#diff-2d6646ee7696c1f5a6469d91f29f50bc67af5669d09833eaba3ea0e0611d80bcR1-R35)

### Storybook Configuration Updates:

* [`web/libs/storybook/.storybook/manager-head.html`](diffhunk://#diff-e4d7b47bafe0069fda4001d19bb35a933930da1febced01f856f73ab3d1899e0R1-R6): Added CSS to set the background of the Storybook preview to transparent.
* [`web/libs/storybook/.storybook/manager-head.prod.html`](diffhunk://#diff-bb7cc80770565926e97008dafc2c0aa787c1e04c3abf6a7058bdf1657eaeb57aR2-R8): Added CSS to set the background of the Storybook preview to transparent in production.
* [`web/libs/storybook/.storybook/preview-head.html`](diffhunk://#diff-480ffabeaf55cb3c6d8c285a0b51b5599f04f1e58d2fcf7a89855b4c9bef0773R1-R13): Added a script to handle the theme color based on user preferences.
* [`web/libs/storybook/.storybook/preview-head.prod.html`](diffhunk://#diff-640b04b63f5cba46bdc8f9261b58df42697369a861f63da29af3f7a6633ac547R2-R14): Added a script to handle the theme color based on user preferences in production.
* [`web/libs/storybook/.storybook/preview.scss`](diffhunk://#diff-7f3884ec078778c42effc49e0083b2df9d0b3d7f8f723bfc9db633273ba30667R5-R12): Updated styles to use CSS variables for color and background.
* [`web/libs/storybook/.storybook/preview.ts`](diffhunk://#diff-ae6b99cdf27eb8bbe2337588aaf71a96ef936c30993ae56c66986a85e7724608R3): Imported the theme toggle preview script.

### Color Tokens Update:

* [`web/libs/ui/src/tokens/tokens.scss`](diffhunk://#diff-7972d0ac6fb81f742daba2cb1c61f92f8f9fd86c7d693d0753c7a8ed26a37559R5-R20): Added raw values for various color tokens to support the new theme toggle functionality. [[1]](diffhunk://#diff-7972d0ac6fb81f742daba2cb1c61f92f8f9fd86c7d693d0753c7a8ed26a37559R5-R20) [[2]](diffhunk://#diff-7972d0ac6fb81f742daba2cb1c61f92f8f9fd86c7d693d0753c7a8ed26a37559R32-R43) [[3]](diffhunk://#diff-7972d0ac6fb81f742daba2cb1c61f92f8f9fd86c7d693d0753c7a8ed26a37559R86-R102) [[4]](diffhunk://#diff-7972d0ac6fb81f742daba2cb1c61f92f8f9fd86c7d693d0753c7a8ed26a37559R113-R129) [[5]](diffhunk://#diff-7972d0ac6fb81f742daba2cb1c61f92f8f9fd86c7d693d0753c7a8ed26a37559R140-R156) [[6]](diffhunk://#diff-7972d0ac6fb81f742daba2cb1c61f92f8f9fd86c7d693d0753c7a8ed26a37559R167-R265) [[7]](diffhunk://#diff-7972d0ac6fb81f742daba2cb1c61f92f8f9fd86c7d693d0753c7a8ed26a37559R482-R497) [[8]](diffhunk://#diff-7972d0ac6fb81f742daba2cb1c61f92f8f9fd86c7d693d0753c7a8ed26a37559R509-R520)

bmartel committed:
chore: OPTIC-1935: Generate all required css variables from design tokens

bmartel committed:
fixing storybook stories to properly allow a global theme toggle

bmartel committed:
fix icon and token stories colors

bmartel committed:
add more semantic tokens

bmartel committed:
adding remaining typography tokens

bmartel committed:
add to storybook

yyassi-heartex reviewed (approved):
there is alot there but i didn't see anything that jumped out at me as odd, I love to see the move away from inline styles towards using classes + putting the atom definitions in to a better spot

nick-skriabin commented on web/libs/storybook/addons/theme-toggle/theme-toggle.tsx:
```diff
@@ -0,0 +1,35 @@
+    setTheme((previousTheme) => THEMES[(THEMES.indexOf(previousTheme) + 1) % THEMES.length]);
+  }, []);
+
+  React.useEffect(() => {
```
```suggestion
  useEffect(() => {
```

for the sake of consistency

nick-skriabin commented on web/libs/storybook/addons/theme-toggle/theme-toggle.tsx:
```diff
@@ -0,0 +1,35 @@
+import React from "react";
```
```suggestion
import { useEffect, useCallback } from "react";
```

bmartel replied:
Funny enough for it to be built properly it needs React in the atmospheric imports, so I just kept this like this otherwise biome removes it 😆

nick-skriabin commented on web/libs/storybook/addons/theme-toggle/theme-toggle.tsx:
```diff
@@ -0,0 +1,35 @@
+  const evaluatedTheme = useAtomValue(evaluatedThemeAtom);
+  const api = useStorybookApi();
+
+  const toggleTheme = React.useCallback(() => {
```
```suggestion
  const toggleTheme = useCallback(() => {
```

bmartel committed:
makeshift an auto icon for storybook theme toggle

bmartel commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14360441113)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-optic-1935'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14360441113

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14360441113
